internal/gmm/handler.go

```diff
@@ -1210,20 +1212,46 @@ func handleRequestedNssai(ue *context.AmfUe, anType models.AccessType) error {
 				// Condition (A) Step 7: initial AMF find Target AMF via NRF ->
 				// Send Namf_Communication_N1MessageNotify to Target AMF
 				ueContext := consumer.GetConsumer().BuildUeContextModel(ue)
-				registerContext := models.RegistrationContextContainer{
-					UeContext:        &ueContext,
-					AnType:           anType,
-					AnN2ApId:         int32(ue.RanUe[anType].RanUeNgapId),
-					RanNodeId:        ue.RanUe[anType].Ran.RanId,
-					InitialAmfName:   amfSelf.Name,
-					UserLocation:     &ue.Location,
-					RrcEstCause:      ue.RanUe[anType].RRCEstablishmentCause,
-					UeContextRequest: ue.RanUe[anType].UeContextRequest,
-					AnN2IPv4Addr:     ue.RanUe[anType].Ran.Conn.RemoteAddr().String(),
-					AllowedNssai: &models.AllowedNssai{
-						AllowedSnssaiList: ue.AllowedNssai[anType],
-						AccessType:        anType,
-					},
+
+				var registerContext models.RegistrationContextContainer
+				AnN2IPAddr := ue.RanUe[anType].Ran.Conn.RemoteAddr().String()
+				ran_host, _, ran_err := net.SplitHostPort(AnN2IPAddr)
+				if ran_err != nil {
+					logger.GmmLog.Errorf("Can't split AnN2IPAddr %+v", ran_err)
+				}
+				ran_addr := netip.MustParseAddr(ran_host)
```

Comment:
Also, be careful, since you don't return when ran_err != nil, netip.MustParseAddr(ran_host) will panic, which is not OK in the gmm/handler.
Panic/Fatalf is only OK at the initialization steps / when validating the configuration IMHO.

Reply:
You right, I fix that two mistakes, the first condition has now a returns, and I add a second condition to check the ParseAddr of ran_host